website/docs/cli/workspaces/index.mdx

```diff
@@ -7,72 +7,99 @@ description: >-
-completely separate state data is to use multiple
-[working directories](/cli/init) (with different
-[backend](/language/settings/backends/configuration) configurations per directory, if you
-aren't using the default `local` backend).
-
-However, this isn't always the most _convenient_ way to handle separate states.
-Terraform installs a separate cache of plugins and modules for each working
-directory, so maintaining multiple directories can waste bandwidth and disk
-space. You must also update your configuration code from version control
-separately for each directory, reinitialize each directory separately when
-changing the configuration, etc.
-
-Workspaces allow you to use the same working copy of your configuration and the
-same plugin and module caches, while still keeping separate states for each
-collection of resources you manage.
+Workspaces in the Terraform CLI refer to separate instances of [state data](/language/state) inside the same working directory.
+
+Terraform relies on state to associate resources with real-world objects. When you run the same configuration multiple times with separate state data, Terraform can manage multiple sets of non-overlapping resources.
+
+Workspaces let you create many instances of the same infrastructure or inject different [variable values](/language/values/variables) for each instance. For example, you may want to quickly create infrastructure to test changes without affecting the production deployment.
+
+## Managing CLI Workspaces
+
+Every [initialized working directory](/cli/init) starts with one workspace named `default`.
+
+Use the [`terraform workspace list`](/cli/commands/workspace/list), [`terraform workspace new`](/cli/commands/workspace/new), and [`terraform workspace delete`](/cli/commands/workspace/delete) commands to manage the available workspaces in the current working directory.
+
+Use [the `terraform workspace select` command](/cli/commands/workspace/select) to change the currently selected workspace. For a given working directory, you can only select one workspace can be at a time. Most Terraform commands only interact with the currently selected workspace. This includes [provisioning](/cli/run) and [state manipulation](/cli/state).
+
+
+## Use Cases
+
+You can create multiple [working directories](/cli/init) to maintain multiple instances of a configuration with completely separate state data. However, Terraform installs a separate cache of plugins and modules for each working directory, so maintaining multiple directories can waste bandwidth and disk space. This approach also requires extra tasks like updating configuration from version control for each directory separately and reinitializing each directory when you change the configuration. Workspaces are more convenient because they let you create different sets of infrastructure with the same working copy of your configuration and the same plugin and module caches.
```

Comment:
Comparing this to the old version of the page I see what you've done here is hoist the "what are workspaces useful for?" content up to the top, which seems reasonable given that workspaces are what this page is about.

I worry a little that this change in emphasis has subtly changed the subtext, though. The previous version of the page is essentially saying "Most of the time you don't need the extra complexity of workspaces, but there are a few benefits that _might_ apply to your situation." This version starts with "Here are all of the benefits of workspaces!", which makes it sound like we're recommending multiple workspaces as the default way to use Terraform.

I'm not sure what to suggest to resolve this. This new version is far more concise and I like it a lot, so I'm certainly not meaning to suggest we bring back all of the old waffly text.

What I'd like to avoid is readers of this page being left with the impression that workspaces are a crucial part of using Terraform CLI without Terraform Cloud, since I don't think that's really true... instead, they exist primarily to support Terraform Cloud and _may_ be useful in some certain situations without Terraform Cloud.

But for most people workspaces will probably just make Terraform harder to use, because they will now have one extra ambient setting to think about every time they are about to run `terraform apply` and will inevitably also need to start adding extra `-var` and `-var-file` options to provide the workspace-specific input variables. Without workspaces all of that information lives inside the configuration itself and so you can just run `terraform apply` and not worry about it. (For those who are using Terraform Cloud, the Terraform Cloud-specific features remove most of this complexity, and so multiple workspaces is far more useful for Terraform Cloud users.)

I'm sorry this isn't a very actionable comment. If you'd like to talk about this in a more synchronous way then I'd be happy to.

Reply:
no no this is super helpful, Martin, and I think you're right. I really appreciate you calling this out. Let me think through how to fix this. i think I can rework a few strategic things to give users the right impression. You're right - right now it's like YAY WORKSPACES ARE SO. HELPFUL, when in reality they can be helpful in some circumstances and are available to you....but you really should only use them for specific things and they do also come with drawbacks.